google_compute_engine/metadata_scripts/tests/script_retriever_test.py

```diff
@@ -44,6 +44,21 @@ def testDownloadGsNoExec(self, mock_call):
 
   @mock.patch('google_compute_engine.metadata_scripts.script_retriever.subprocess.check_call')
   @mock.patch('google_compute_engine.metadata_scripts.script_retriever.tempfile.NamedTemporaryFile')
+  def testDownloadGsUrlFallback(self, mock_tempfile, mock_call):
```

Comment:
This test should be combined with `testDownloadScript` so that you can test each url type. Currently you have an issue where you are relying on `_DownloadUrl` returning the correct value, but that isn't the function this test is testing, so the mocks probably aren't set up correctly in Python 3.